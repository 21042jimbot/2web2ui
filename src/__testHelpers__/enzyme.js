export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
