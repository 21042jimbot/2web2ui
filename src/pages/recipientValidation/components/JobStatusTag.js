import React from 'react';
import classNames from 'classnames';
import { useHibana } from 'src/context/HibanaContext';
import { Error, CheckCircle, Cached, CloudUpload } from '@sparkpost/matchbox-icons';
import { Tag } from 'src/components/matchbox';

import useHibanaOverride from 'src/hooks/useHibanaOverride';
import OGStyles from './JobStatusTag.module.scss';
import hibanaStyles from './JobStatusTagHibana.module.scss';

function getStatusProps(styles) {
  return {
    error: {
      tagColor: 'red',
      className: styles.Failed,
      icon: Error,
      message: 'Validation Error',
    },
    usage_limit_exceeded: {
      tagColor: 'red',
      className: styles.Failed,
      icon: Error,
      message: 'Validation Error',
    },
    success: {
      tagColor: 'green',
      className: styles.Complete,
      icon: CheckCircle,
      message: 'Complete',
    },
    queued_for_batch: {
      tagColor: '',
      className: styles.Ready,
      icon: CloudUpload,
      message: 'Ready to validate',
    },
    loading: {
      tagColor: '',
      className: styles.Loading,
      icon: Cached,
      message: 'Processing',
    },
  };
}

const JobStatusTag = ({ status }) => {
  const [state] = useHibana();
  const { isHibanaEnabled } = state;
  const styles = useHibanaOverride(OGStyles, hibanaStyles);
  const statusProps = getStatusProps(styles);
  const { tagColor, className, icon: Icon, message } = statusProps[status] || statusProps.loading;

  return (
    <Tag color={isHibanaEnabled ? tagColor : ''} style={{ verticalAlign: 'bottom' }}>
      <span className={classNames(styles.JobStatusTagContent, className)}>
        {!isHibanaEnabled && <Icon />}
        &nbsp;
      </span>
      <span>{message}</span>
    </Tag>
  );
};

export default JobStatusTag;
