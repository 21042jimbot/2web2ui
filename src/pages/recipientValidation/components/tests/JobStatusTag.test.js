import React from 'react';
import { shallow } from 'enzyme';
import JobStatusTag from '../JobStatusTag';

describe('JobStatusTag', () => {
  const subject = (props = {}) => shallow(<JobStatusTag {...props} />);

  it('renders a loading tag', () => {
    expect(subject()).toMatchSnapshot();
  });

  it('renders an error tag', () => {
    expect(subject({ status: 'error' })).toMatchSnapshot();
  });

  it('renders a ready tag', () => {
    expect(subject({ status: 'queued_for_batch' })).toMatchSnapshot();
  });

  it('renders a success tag', () => {
    expect(subject({ status: 'success' })).toMatchSnapshot();
  });
});
