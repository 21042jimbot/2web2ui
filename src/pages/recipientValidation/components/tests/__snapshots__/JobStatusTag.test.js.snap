// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobStatusTag renders a loading tag 1`] = `
<Tag>
  <span
    className="Loading"
  >
    <Cached />
     
  </span>
  <span>
    Processing
  </span>
</Tag>
`;

exports[`JobStatusTag renders a ready tag 1`] = `
<Tag>
  <span
    className="Ready"
  >
    <CloudUpload />
     
  </span>
  <span>
    Ready to validate
  </span>
</Tag>
`;

exports[`JobStatusTag renders a success tag 1`] = `
<Tag>
  <span
    className="Complete"
  >
    <CheckCircle />
     
  </span>
  <span>
    Completed
  </span>
</Tag>
`;

exports[`JobStatusTag renders an error tag 1`] = `
<Tag>
  <span
    className="Failed"
  >
    <Error />
     
  </span>
  <span>
    Failed. Please try again.
  </span>
</Tag>
`;
